hr_edl/src/bin/run_cfr_corr_dist.cc

```diff
@@ -0,0 +1,95 @@
+// This file runs standard CFR on a .efg game.
+//
+// The graph from Sec 6.1 of Morrill et al. 2020, Hindsight and Sequential
+// Rationality of Correlated Play (https://arxiv.org/abs/2012.05874) can be
+// reproduced by running this code on efg/extended_shapleys.efg.
+
+#include <memory>
+#include <queue>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "absl/flags/usage.h"
+#include "absl/strings/match.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_join.h"
+#include "open_spiel/algorithms/cfr.h"
+#include "open_spiel/algorithms/corr_dev_builder.h"
+#include "open_spiel/algorithms/corr_dist.h"
+#include "open_spiel/game_transforms/turn_based_simultaneous_game.h"
+#include "open_spiel/spiel.h"
+
+// Environment
+ABSL_FLAG(std::string, efg_file, "efg/extended_shapleys.efg",
+          "The EFG file representing the game.");
+ABSL_FLAG(int32_t, t, 1000, "The number of iterations to run.");
+
+inline constexpr int kSeed = 23894982;
+
+// Reporting
+ABSL_FLAG(
+    double, report_gap_factor, 2.0,
+    "The factor increasing the number of iterations to finish between "
+    "reporting exploitability. E.g. 1 prints after each iteration, 2 prints "
+    "after 2^i iterations, i >= 1.");
+ABSL_FLAG(int32_t, report_skip, 0,
+          "Number of initial iterations for which to skip reporting.");
+
+using open_spiel::algorithms::CorrDevBuilder;
+using open_spiel::algorithms::CFRSolverBase;
+using open_spiel::algorithms::CorrDistConfig;
+using open_spiel::TabularPolicy;
+using open_spiel::algorithms::CorrelationDevice;
+using open_spiel::algorithms::AFCCEDist;
+using open_spiel::algorithms::AFCEDist;
+using open_spiel::algorithms::EFCCEDist;
+using open_spiel::algorithms::EFCEDist;
+using open_spiel::algorithms::ExpectedValues;
+using open_spiel::algorithms::CFRCurrentPolicy;
+
+void run_experiment() {
+  const std::shared_ptr<const open_spiel::Game> game =
+      open_spiel::LoadGameAsTurnBased(
+          "efg_game", {{"filename", open_spiel::GameParameter(
+                                        absl::GetFlag(FLAGS_efg_file))}});
+  const int iterations = absl::GetFlag(FLAGS_t);
+
+  CorrDevBuilder cd_builder;
+  CFRSolverBase solver(*game,
+                       /*alternating_updates=*/false,
```

Comment:
Did you want to add an option to run with alternating updates since we have a figure for it in the paper?

Reply:
Yep done (same with random initial regrets)